plugins/meta/sbr/README.md

```diff
@@ -0,0 +1,128 @@
+applications which are using network separation for traffic management and
+security, there is no way to tell *a priori* which interface should be used,
+but the application is capable of making the decision.
+
+As an example, a Telco application might have two networks, a management
+network and a SIP (telephony) network for traffic, with rules which state that:
+
+- SIP traffic (only) must be routed over the SIP network;
+
+- all other traffic (but no SIP traffic) must be routed over the management
+  network.
+
+There is no way of configuring this based on destination IP, since there is no
+way of telling whether a destination IP on the internet is (say) an address
+used for downloading updated software packages or a remote SIP endpoint.
+
+Hence Source Based Routing is used.
+
+- The application explicitly listens on the correct interface for incoming
+  traffic.
+
+- When the application wishes to send to an address via the SIP network, it
+  explicitly binds to the IP of the device on that network.
+
+- Routes for the SIP interface are configured in a separate routing table, and
+  a rule is configured to use that table based on the source IP address.
+
+Note that in most cases there is a management device (the first one) and that
+the SBR plugin is called once for each device after the first, leaving the
+default routing table applied to the management device. However, this not
+mandatory, and source based routing may be configured on any or all of the
+devices as appropriate.
+
+## Usage
+
+This plugin runs as a chained plugin, and requires the following information
+passed in from the previous plugin (which has just set up the network device):
+
+- What is the network interface in question?
+
+- What is the IP address (or addresses) of that network interface?
+
+- What is the default gateway for that interface (if any)?
+
+It then reads all routes to the network interface in use.
+
+Here is an example of what the plugin would do. (The `ip` based commands are
+implemented in go, but easier to describe via the command line.) Suppose that
+it reads that:
+
+- The interface is `net1`.
+
+- The IP address on that interface is `192.168.1.209`.
+
+- The default gateway on that interface is `192.168.1.1`.
+
+- There is one route configured on that network interface, which is
+  `192.168.1.0/24`.
+
+Then the actions it takes are the following.
+
+- It creates a new routing table, and sets a rule to use it for the IP address in question.
+
+        ip rule add from 192.168.1.209/32 table 100
+
+- It adds a route to the default gateway for the relevant table.
+
+        ip route add default via 192.168.1.1 table 100
+
+- It moves every existing route on the device to the new table.
+
+        ip route del 192.168.1.0/24 dev net1 src 192.168.1.209
+        ip route add 192.168.1.0/24 dev net1 src 192.168.1.209 table 100
+
+On deletion it does nothing at all (since it assumes the pod is about to be
```

Comment:
With Kubernetes as a runtime you can assume this, but CNI isn't only used with Kubernetes and so I don't think we can assume this in the general case.  Would it be hard to add cleanup on DEL?

Reply:
Good point. I think it's probably fairly easy to tidy up the table and delete the rule that sends to it. I'll tweak the PR in due course, and update the docs to reflect.